src/snowflake/snowpark/modin/plugin/extensions/rolling_groupby_overrides.py

```diff
@@ -0,0 +1,333 @@
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+# Code in this file may constitute partial or total reimplementation, or modification of
+# existing code originally distributed by the Modin project, under the Apache License,
+# Version 2.0.
+
+# Code in this file may constitute partial or total reimplementation, or modification of
+# existing code originally distributed by the pandas project, under the BSD 3-Clause License
+
+
+"""Implement RollingGroupby public API."""
+from typing import Any, Union
+
+import modin.pandas as pd
+
+
+from snowflake.snowpark.modin.plugin._internal.telemetry import TelemetryMeta
+from snowflake.snowpark.modin.plugin.utils.error_message import (
+    ErrorMessage,
+    series_not_implemented,
+)
+from snowflake.snowpark.modin.plugin.utils.warning_message import WarningMessage
+from .series_overrides import register_series_accessor
+from pandas._libs import lib
+from pandas._typing import AnyArrayLike
+
+
+class RollingGroupby(metaclass=TelemetryMeta):
+    def __init__(
+        self,
+        dataframe,
+        by,
+        window,
+        min_periods: Union[int, None] = None,
+        center: bool = False,
+        win_type: Union[str, None] = None,
+        on: Union[str, None] = None,
+        axis: Union[int, str] = 0,
+        closed: Union[str, None] = None,
+        method: str = "single",
+        dropna: bool = True,
+    ) -> None:
+        # TODO: SNOW-1063349: Modin upgrade - modin.pandas.groupby.DataFrameGroupBy functions
+        self._dataframe = dataframe
+        self._query_compiler = dataframe._query_compiler
+        self.by = by
+        self.rolling_kwargs = {
+            "window": window,
+            "min_periods": min_periods,
+            "center": center,
+            "win_type": win_type,
+            "on": on,
+            "axis": axis,
+            "closed": closed,
+            "method": method,
+        }
+        self.groupby_kwargs = {
+            "by": by,
+            "dropna": dropna,
+        }
+        self._series_not_implemented()
+
+    def _method_not_implemented(self, method: str):  # pragma: no cover
+        # TODO: SNOW-1063349: Modin upgrade - modin.pandas.groupby.DataFrameGroupBy functions
+        raise ErrorMessage.not_implemented(
```

Comment:
```suggestion
        ErrorMessage.not_implemented(
```

`not_implemented` will `raise` for you.